src/main/java/io/orangebeard/listener/OrangebeardTestSystemListener.java

```diff
@@ -187,32 +191,36 @@ private UUID getAndOrStartSuite(TestPage testPage) {
             suitePath = suitePath + "." + suite;
             suiteId = runContext.getSuiteId(suitePath);
             if (suiteId == null) {
+                String description = null;
                 Set<Attribute> suiteAttrs = null;
-                if(suitePath.endsWith(fullSuiteName)) {
-                     suiteAttrs = retrieveSuiteMetaData(testPage);
+                PageData suitePageData = getPageDataForSuite(suitePath, testPage.getSourcePage());
+                if (suitePageData != null && suitePageData.getAttribute(WikiPageProperty.SUITES) != null) {
+                    suiteAttrs = extractTags(suitePageData.getAttribute(WikiPageProperty.SUITES));
                 }
-                suiteId = startSuite(parentSuiteId, suite, suiteAttrs);
+                if (suitePageData != null && suitePageData.getAttribute(WikiPageProperty.HELP) != null) {
+                    description = suitePageData.getAttribute(WikiPageProperty.HELP);
+                }
+
+                suiteId = startSuite(parentSuiteId, suite, description, suiteAttrs);
                 runContext.addSuite(suitePath, suiteId);
             }
         }
 
         return suiteId;
     }
 
-    private Set<Attribute> retrieveSuiteMetaData(TestPage testPage) {
-        Set<Attribute> metaData = new HashSet<>();
-        if (testPage instanceof WikiTestPage) {
-
-            PageData suitePageData = ((WikiTestPage) testPage).getSourcePage().getParent().getData();
-            if (suitePageData.getAttribute(WikiPageProperty.SUITES) != null) {
-                metaData.addAll(extractTags(suitePageData.getAttribute(WikiPageProperty.SUITES)));
-            }
+    private PageData getPageDataForSuite(String suitePath, WikiPage sourcePage) {
```

Comment:
deze methode is echt magisch voor mij; misschien moet ie wat betere naamgeving of een comment krijgen? Of ligt het aan mij?

Reply:
Het kreng zoekt het PageData object op voor de suite die je meegeeft in de wikipage van de test die de suite triggert (de wikipage bevat ook alle parent page objecten) Vond de naam wel dat omvatten :)

Zal een commentje maken. Gaat uit van bekendheid van (quirky) FitNesse model